### PR TITLE
[tools/pareto_profiler] Implement HLPS algorithm for pareto estimation 

### PR DESCRIPTION
**Pull request by dr-venkman**
Implement HLPS (Hamming Local Pareto Search) algorithm to jointly estimate best possible execution time and peak memory consumption for on-device inference models.

ONE-DCO-1.0-Signed-off-by: venkat.iyervenkat.iyer@samsung.com

### CODE REVIEW

File: tools/pareto_profiler/estimator/Hlps.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
+#! /usr/bin/python
```
Comment (intom) [inline tools/pareto_profiler/estimator/Hlps.py:1]: I think it needs to add copyright.

**Review by intom — COMMENTED** [review]
please add copyright

File: tools/pareto_profiler/estimator/Hlps.py (hunk 1 above)
Comment (dr-venkman) [reply tools/pareto_profiler/estimator/Hlps.py:1]: will add it, thanks for spotting.

**Review by intom — APPROVED** [review]
LGTM

**Review by hseok-oh — APPROVED** [review]

**Pull request merged**